File: crates/travsr-plugin-host/tests/fixtures/Hello.java
package com.example;

import java.util.List;
import java.util.ArrayList;

public class PaymentService {
    private final String apiKey;

    public PaymentService(String apiKey) {
        this.apiKey = apiKey;
    }

    public boolean charge(double amount) {
        return apiKey != null && !apiKey.isEmpty() && amount > 0;
    }

    public List<String> getTransactions() {
        return new ArrayList<>();
    }
}

File: crates/travsr-plugin-host/tests/fixtures/Hello.kt
package com.example

import kotlin.collections.List

class PaymentService(private val apiKey: String) {

    fun charge(amount: Double): Boolean {
        return apiKey.isNotEmpty() && amount > 0
    }

    fun getTransactions(): List<String> = emptyList()
}

fun processPayment(service: PaymentService, amount: Double) = service.charge(amount)
